### PR TITLE
Handle non-finite floats in metrics

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: lib/axon/loop.ex [hunk 1]
```diff
@@ -556,13 +556,16 @@ defmodule Axon.Loop do
...
-      :bf -> "#{name}: #{:io_lib.format('~.3f', [Nx.to_number(val)])}"
-      _ -> "Unsupported type of metric"
+      t when t in [:s, :u] -> "#{name}: #{Nx.to_number(val)}"
+      :f -> "#{name}: #{float_format('~.7f', Nx.to_number(val))}"
+      :bf -> "#{name}: #{float_format('~.3f', Nx.to_number(val))}"
+      _ -> "#{name}: unsupported type of metric #{inspect(type)}"
     end
   end
 
+  defp float_format(format, val) when is_atom(val), do: Atom.to_string(val)
```
Comment (polvalente) [inline lib/axon/loop.ex:566]: In Nx and Complex we inspect :nan, :infinity and :neg_infinity respectively as NaN, Inf and -Inf

Shouldn't we do the same here?

File: lib/axon/loop.ex [hunk 2]
```diff
@@ -556,13 +556,18 @@ defmodule Axon.Loop do
...
+      t when t in [:s, :u] -> "#{name}: #{Nx.to_number(val)}"
+      :f -> "#{name}: #{float_format('~.7f', Nx.to_number(val))}"
+      :bf -> "#{name}: #{float_format('~.3f', Nx.to_number(val))}"
+      _ -> "#{name}: unsupported type of metric #{inspect(type)}"
     end
   end
 
+  defp float_format(_format, :nan) when is_atom(val), do: "NaN"
+  defp float_format(_format, :infinity) when is_atom(val), do: "Inf"
+  defp float_format(_format, :neg_infinity) when is_atom(val), do: "-Inf"
```
Comment (josevalim) [inline lib/axon/loop.ex:568]:
```suggestion
  defp float_format(_format, :nan), do: "NaN"
  defp float_format(_format, :infinity), do: "Inf"
  defp float_format(_format, :neg_infinity), do: "-Inf"
```

File: lib/axon/loop.ex [hunk 3]
```diff
@@ -556,13 +556,18 @@ defmodule Axon.Loop do
...
+      :f -> "#{name}: #{float_format('~.7f', Nx.to_number(val))}"
+      :bf -> "#{name}: #{float_format('~.3f', Nx.to_number(val))}"
+      _ -> "#{name}: unsupported type of metric #{inspect(type)}"
     end
   end
 
+  defp float_format(_format, :nan), do: "NaN"
+  defp float_format(_format, :infinity), do: "Inf"
+  defp float_format(_format, :neg_infinity), do: "-Inf"
+  defp float_format(format, val) when is_atom(val), do: :io_lib.format(format, [val])
```
Comment (alisinabh) [inline lib/axon/loop.ex:569]: Maybe `when not is_atom(val)`?

Comment (josevalim) [inline lib/axon/loop.ex:569]:
```suggestion
  defp float_format(format, val) when is_float(val), do: :io_lib.format(format, [val])
```

**Review by seanmor5 — APPROVED** [review]

**Pull request merged**